src/modules/esl-utils/misc/array.ts

```diff
@@ -29,13 +29,34 @@ export const uniq = <T> (arr: T[]): T[] => {
   return result;
 };
 
-/** Crete an array filled with the range [0,..,N-1] */
+/** Create an array filled with the range [0,..,N-1] */
 export function range(n: number): number[];
-/** Crete an array filled with values returned by the filler callback */
+/** Create an array filled with values returned by the filler callback */
 export function range<T>(n: number, filler: (i: number) => T): T[];
 export function range(n: number, filler: (i: number) => any = identity): any[] {
   const arr = Array(n);
   let i = 0;
   while (i < n) arr[i] = filler(i++);
   return arr;
 }
+
+/** Create an array of unique values from two arrays*/
+export function intersection(a: any, b: any): any[] {
+  const arr = new Set([...a, ...b]);
+  return [...arr];
+}
+
+/** Create an array from two arrays*/
+export function union(a: any, b: any): any[] {
+  return [...a, ...b];
```

Comment:
1) Incorrect implementation sounds like the current intersection implementation is union:
`expect(union([1, 2], [1, 2, 3])).toEqual([1, 2, 3])`
2) It will be better to make method support more then array pair, and accept unlimited number of initial arrays